projects/pgai/pgai/logger.py

```diff
@@ -0,0 +1,71 @@
+import json
+import logging
+from collections.abc import Callable
+from typing import Any, TypeVar
+
+from typing_extensions import override
+
+T = TypeVar("T", bound="StructuredMessage")
+
+# Type for the renderer function
+RendererType = Callable[[str, dict[str, Any]], str]
+
+
+class StructuredMessage:
```

Comment:
I don't understand why we're introducing the `StructuredMessage` class. I am aware that we had structured logging with structlog, but I don't understand if we're just trying to replicate what structlog had, or provide some functionality for our users.

Reply:
This is really mostly for internal usage to be able to keep some of the structured data we already specified in the output